app/controllers/people.js

```diff
@@ -1,39 +1,61 @@
 import { Person } from '../models';
 import { Authenticate } from '../services';
+import { error } from '../utils';
 
 export default class PeopleController {
-    static index() {
-        return Person.fetchAll();
+    static async index(req, res) {
+        const people = await Person.fetchAll();
+        return res.status(200).json(people);
     }
 
-    static show(id) {
-        return Person.find(id);
+    static async show(req, res) {
+        const { id } = req.params;
+        const person = await Person.find(id);
+        if (person) {
+            return res.status(200).json(person);
+        } else {
+            return res.status(404).json(error(`Person with ID ${id}, cannot be found.`));
+        }
     }
 
-    static update(id, data) {
-        return Person.find(id)
-            .then(person => {
-                if (!person) return false
-
-                return person.save(data, { patch: true });
-            });
+    static async update(res, req) {
+        const { id } = req.params;
+        const { body } = req.body;
+        const person = await Person.find(id);
+        
+        if (person) {
+            const response = await person.save(body, { patch: true });
+            if (response) return res.status(204).end();
+        } else {
+            return res.status(404).json(error(`Person with ID ${id}, cannot be found.`));
+        }
     }
 
-    static create(data) {
-        const { password, ...model } = data;
+    static async create(req, res) {
+        const { password, ...model } = req.body;
+        
         if (password) {
-            return Authenticate.hash(password)
-                .then(password => {
-                    const person = new Person({ password, ...model });
-
-                    return person.save();
-                });
+            const pass = await Authenticate.hash(password);
+            try {
+                const person = await new Person({ password: pass, ...model }).save();
+                return res.status(201).end();
+            } catch (e) {
+                return res.status(404).json(error(e.stack, e));
+            }
         }
-
-        return Promise.reject();
+        
+        return res.status(404).json(error(`Password is required`));
```

Comment:
Not really. we weren't allowing person create in laravel so there wasn't really a set handling if the password is wrong or not present. So till then.